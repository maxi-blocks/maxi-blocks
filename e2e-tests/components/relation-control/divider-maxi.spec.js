/**
 * WordPress dependencies
 */
import {
	createNewPost,
	pressKeyWithModifier,
	selectBlockByClientId,
	saveDraft,
} from '@wordpress/e2e-test-utils';

/**
 * Internal dependencies
 */
import {
	openSidebarTab,
	editColorControl,
	getAttributes,
	openPreviewPage,
	insertMaxiBlock,
} from '../../utils';

describe('Divider Maxi hover simple actions', () => {
	beforeEach(async () => {
		await createNewPost();
		await insertMaxiBlock(page, 'Divider Maxi');
		await page.evaluate(() => {
			// Get the client ID of the currently selected block
			const clientId = wp.data
				.select('core/block-editor')
				.getSelectedBlockClientId();

			// Check if a block is selected
			if (clientId) {
				// Set the new uniqueID value
				const newValue = 'divider-maxi-1se8ef1z-u';

				// Update the block's uniqueID attribute
				wp.data
					.dispatch('core/block-editor')
					.updateBlockAttributes(clientId, { uniqueID: newValue });
			}
		});
		// Add native paragraph block
		await selectBlockByClientId(
			await page.$eval('.maxi-divider-block', el =>
				el.getAttribute('data-block')
			)
		);
		await page.keyboard.press('Enter');

		await insertMaxiBlock(page, 'Button Maxi');
		await page.evaluate(() => {
			// Get the client ID of the currently selected block
			const clientId = wp.data
				.select('core/block-editor')
				.getSelectedBlockClientId();

			// Check if a block is selected
			if (clientId) {
				// Set the new uniqueID value
				const newValue = 'button-maxi-1se8ef1z-u';

				// Update the block's uniqueID attribute
				wp.data
					.dispatch('core/block-editor')
					.updateBlockAttributes(clientId, { uniqueID: newValue });
			}
		});
		await openSidebarTab(page, 'advanced', 'interaction builder');

		// Add interaction
		await page.waitForSelector('.maxi-relation-control__button');
		await page.$eval('.maxi-relation-control__button', el => el.click());

		// Add title
		const textControls = await page.$$('.maxi-text-control__input');
		await textControls[1].focus();
		await page.keyboard.type('Hello World!');
		await page.waitForTimeout(150);

		// Add target
		let selectControls = await page.$$('.maxi-select-control__input');
		await selectControls[1].select('divider-maxi-1se8ef1z-u');

		// Add action
		selectControls = await page.$$('.maxi-select-control__input');
		await selectControls[2].select('hover');
	});

	const checkFrontend = async () => {
		await saveDraft();

		const previewPage = await openPreviewPage(page);
		await previewPage.waitForSelector('.entry-content');

		await previewPage.waitForSelector(
			'#button-maxi-1se8ef1z-u .maxi-button-block__button'
		);
		await previewPage.hover(
			'#button-maxi-1se8ef1z-u .maxi-button-block__button'
		);
		await previewPage.waitForTimeout(100);

		await previewPage.waitForSelector(
			'#relations--divider-maxi-1se8ef1z-u-styles'
		);
		const stylesCSS = await previewPage.$eval(
			'#relations--divider-maxi-1se8ef1z-u-styles',
			el => el.textContent
		);
		expect(stylesCSS).toMatchSnapshot();

		await previewPage.waitForSelector(
			'#relations--divider-maxi-1se8ef1z-u-in-transitions'
		);
		const inTransitionsCSS = await previewPage.$eval(
			'#relations--divider-maxi-1se8ef1z-u-in-transitions',
			el => el.textContent
		);
		expect(inTransitionsCSS).toMatchSnapshot();

		await previewPage.mouse.move(0, 0);

		await previewPage.waitForSelector(
			'#relations--divider-maxi-1se8ef1z-u-out-transitions'
		);
		const outTransitionsCSS = await previewPage.$eval(
			'#relations--divider-maxi-1se8ef1z-u-out-transitions',
			el => el.textContent
		);
		expect(outTransitionsCSS).toMatchSnapshot();
	};

	it('Divider shadow', async () => {
		// Select setting
		const selectControls = await page.$$('.maxi-select-control__input');
		await selectControls[3].select('dbs');

		// Select first default
		await page.$$eval('.maxi-default-styles-control__button', buttons =>
			buttons[1].click()
		);

		// Shadow color
		await editColorControl({
			page,
			instance: await page.$('.maxi-color-control'),
			paletteStatus: true,
			colorPalette: 8,
			opacity: 11,
		});

		// Shadow horizontal offset
		await page.$$eval(
			'.maxi-advanced-number-control .maxi-advanced-number-control__value',
			ANCs => ANCs[1].focus()
		);
		await pressKeyWithModifier('primary', 'a');
		await page.keyboard.type('22');

		// Shadow vertical offset
		await page.$$eval(
			'.maxi-advanced-number-control .maxi-advanced-number-control__value',
			ANCs => ANCs[2].focus()
		);
		await pressKeyWithModifier('primary', 'a');
		await page.keyboard.type('33');

		// Shadow blur
		await page.$$eval(
			'.maxi-advanced-number-control .maxi-advanced-number-control__value',
			ANCs => ANCs[3].focus()
		);
		await pressKeyWithModifier('primary', 'a');
		await page.keyboard.type('44');

		// Shadow spread
		await page.$$eval(
			'.maxi-advanced-number-control .maxi-advanced-number-control__value',
			ANCs => ANCs[4].focus()
		);
		await pressKeyWithModifier('primary', 'a');
		await page.keyboard.type('55');

		expect(await getAttributes('relations')).toMatchSnapshot();

		await checkFrontend();
	});

	it('Line settings', async () => {
		// Select setting
		const selectControls = await page.$$('.maxi-select-control__input');
		await selectControls[3].select('dls');

		// Select second default
		await page.$$eval('.maxi-default-styles-control__button', buttons =>
			buttons[2].click()
		);

		// Line color
		await editColorControl({
			page,
			instance: await page.$('.maxi-color-control'),
			paletteStatus: true,
			colorPalette: 8,
			opacity: 50,
		});

		// Line size
		await page.$$eval(
			'.maxi-advanced-number-control .maxi-advanced-number-control__value',
			ANCs => ANCs[1].focus()
		);
		await pressKeyWithModifier('primary', 'a');
		await page.keyboard.type('11');

		// Line weight
		await page.$$eval(
			'.maxi-advanced-number-control .maxi-advanced-number-control__value',
			ANCs => ANCs[2].focus()
		);
		await pressKeyWithModifier('primary', 'a');
		await page.keyboard.type('22');

		expect(await getAttributes('relations')).toMatchSnapshot();

		await checkFrontend();
	});
});
