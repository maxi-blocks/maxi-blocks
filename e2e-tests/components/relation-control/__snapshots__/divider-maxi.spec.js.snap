// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Maxi hover simple actions Divider shadow 1`] = `
[
  {
    "action": "hover",
    "attributes": {
      "divider-box-shadow-blur-general": 44,
      "divider-box-shadow-blur-unit-general": "px",
      "divider-box-shadow-blur-unit-xl": "px",
      "divider-box-shadow-blur-xl": 44,
      "divider-box-shadow-horizontal-general": 22,
      "divider-box-shadow-horizontal-unit-general": "px",
      "divider-box-shadow-horizontal-unit-xl": "px",
      "divider-box-shadow-horizontal-xl": 22,
      "divider-box-shadow-inset-general": false,
      "divider-box-shadow-inset-xl": false,
      "divider-box-shadow-palette-color-general": 8,
      "divider-box-shadow-palette-color-xl": 8,
      "divider-box-shadow-palette-opacity-general": 0.11,
      "divider-box-shadow-palette-opacity-xl": 0.11,
      "divider-box-shadow-palette-sc-status-general": false,
      "divider-box-shadow-palette-sc-status-xl": false,
      "divider-box-shadow-palette-status-general": true,
      "divider-box-shadow-palette-status-xl": true,
      "divider-box-shadow-spread-general": 55,
      "divider-box-shadow-spread-unit-general": "px",
      "divider-box-shadow-spread-unit-xl": "px",
      "divider-box-shadow-spread-xl": 55,
      "divider-box-shadow-vertical-general": 33,
      "divider-box-shadow-vertical-unit-general": "px",
      "divider-box-shadow-vertical-unit-xl": "px",
      "divider-box-shadow-vertical-xl": 33,
    },
    "css": {
      "general": {
        "breakpoint": null,
        "styles": {
          "box-shadow": "22px 33px 44px 55px rgba(var(--maxi-light-color-8,150,176,203),0.11)",
        },
      },
    },
    "effects": {
      "disableTransition": false,
      "easing-general": "ease",
      "hoverStatus": false,
      "transition-delay-general": 0,
      "transition-duration-general": 0.3,
      "transition-status-general": true,
      "transitionTarget": " .maxi-divider-block__divider",
    },
    "id": 1,
    "isButton": true,
    "sid": "dbs",
    "target": " .maxi-divider-block__divider",
    "title": "Hello World!",
    "uniqueID": "divider-maxi-1",
  },
]
`;

exports[`Divider Maxi hover simple actions Divider shadow 2`] = `" body.maxi-blocks--active #divider-maxi-1[data-maxi-relations=\"true\"] .maxi-divider-block__divider { box-shadow: 22px 33px 44px 55px rgba(var(--maxi-light-color-8,150,176,203),0.11);}"`;

exports[`Divider Maxi hover simple actions Divider shadow 3`] = `"body.maxi-blocks--active #divider-maxi-1[data-maxi-relations=\"true\"] .maxi-divider-block__divider { transition: all 0s ease 0s, box-shadow 0.3s ease 0s;}"`;

exports[`Divider Maxi hover simple actions Divider shadow 4`] = `"body.maxi-blocks--active #divider-maxi-1[data-maxi-relations=\"true\"] .maxi-divider-block__divider { transition: all 0s ease 0s, box-shadow 0.3s ease 0s;}"`;

exports[`Divider Maxi hover simple actions Line settings 1`] = `
[
  {
    "action": "hover",
    "attributes": {
      "divider-border-palette-color-general": 8,
      "divider-border-palette-color-xl": 8,
      "divider-border-palette-opacity-general": 0.5,
      "divider-border-palette-opacity-xl": 0.5,
      "divider-border-palette-sc-status-general": false,
      "divider-border-palette-sc-status-xl": false,
      "divider-border-palette-status-general": true,
      "divider-border-palette-status-xl": true,
      "divider-border-style-general": "dashed",
      "divider-border-style-xl": "dashed",
      "divider-border-top-width-general": 22,
      "divider-border-top-width-xl": 22,
      "divider-width-general": 11,
      "divider-width-unit-general": "%",
      "divider-width-xl": 11,
    },
    "css": {
      "general": {
        "breakpoint": null,
        "styles": {
          "border-color": "var(--maxi-light-divider-color,rgba(var(--maxi-light-color-8,150,176,203),0.5))",
          "border-right-style": "none",
          "border-top-style": "dashed",
          "border-top-width": "22px",
          "height": "22px",
          "width": "11%",
        },
      },
    },
    "effects": {
      "disableTransition": false,
      "easing-general": "ease",
      "hoverStatus": false,
      "transition-delay-general": 0,
      "transition-duration-general": 0.3,
      "transition-status-general": true,
    },
    "id": 1,
    "isButton": true,
    "sid": "dls",
    "target": " .maxi-divider-block__divider",
    "title": "Hello World!",
    "uniqueID": "divider-maxi-1",
  },
]
`;

exports[`Divider Maxi hover simple actions Line settings 2`] = `" body.maxi-blocks--active #divider-maxi-1[data-maxi-relations=\"true\"] .maxi-divider-block__divider { width: 11%; height: 22px; border-top-width: 22px; border-right-style: none; border-top-style: dashed; border-color: var(--maxi-light-divider-color,rgba(var(--maxi-light-color-8,150,176,203),0.5));}"`;

exports[`Divider Maxi hover simple actions Line settings 3`] = `"body.maxi-blocks--active #divider-maxi-1[data-maxi-relations=\"true\"] .maxi-divider-block__divider { transition: all 0s ease 0s, border-color 0.3s ease 0s, border-top-style 0.3s ease 0s, border-right-style 0.3s ease 0s, border-top-width 0.3s ease 0s, height 0.3s ease 0s, width 0.3s ease 0s;}"`;

exports[`Divider Maxi hover simple actions Line settings 4`] = `"body.maxi-blocks--active #divider-maxi-1[data-maxi-relations=\"true\"] .maxi-divider-block__divider { transition: all 0s ease 0s, border-color 0.3s ease 0s, border-top-style 0.3s ease 0s, border-right-style 0.3s ease 0s, border-top-width 0.3s ease 0s, height 0.3s ease 0s, width 0.3s ease 0s;}"`;
